### ISSUE TITLE
Add SAS syntax highlighting

### ISSUE THREAD

**Issue by Mr-ZBin** [microsoft/monaco-editor#2613]
Hi,

I'm a SAS user, and would you mind adding SAS syntax highlighting? 

SAS (full name: statistical analysis system, referred to as SAS) is one of the largest private software companies in the world. It is a statistical analysis software developed by North Carolina State University in 1966.

Code looks like below screenshot:
![image](https://user-images.githubusercontent.com/22544491/129180677-050f6879-9244-460b-8795-e858786364b6.png)

Thanks

**Comment by hediet** [comment]
SAS is on place 21 on the TIOBE index, so I think its relevance definitely justifies including a grammar for it.

We are very open for a PR on this!

Please have a look at how other languages are defined in https://github.com/microsoft/monaco-languages.

**Comment by Mr-ZBin** [comment]
Hi,

Do we need to custom syntax highlighting on SAS language by ourselves? The monaco editor has been supporting 77 basic-languages but not include the SAS language .

Thanks.

**Comment by allanbowe** [comment]
My team are keen to build this extension, so that we can offer syntax highlighting in https://github.com/sasjs/server, however - we need sponsorship

**Comment by Mr-ZBin** [comment]
Hi,

What can I do for the sponsorship? Actually, I'm not famililar with javascript.

**Comment by allanbowe** [comment]
@Mr-ZBin  - we have JavaScript developers, just looking for a project or company that is willing to share the resource cost (which we can subsidise) of building the feature.

Happy to discuss more - feel free to contact us on:  https://sasapps.io/contact

**Comment by XiaoPengYouzzZ** [comment]
@Mr-ZBin Hello, I want to know if you have finally completed this task, integrating the sas language into the monaco editor. If not, do you have any other solutions? I'm currently facing this challenge as well.

**Comment by Mr-ZBin** [comment]
> @Mr-ZBin Hello, I want to know if you have finally completed this task, integrating the sas language into the monaco editor. If not, do you have any other solutions? I'm currently facing this challenge as well.

No, I use VSCode with Official SAS Language Extension at present.

**Comment by XiaoPengYouzzZ** [comment]
@Mr-ZBin Okay, thank you for your reply. I tried to do this work in the custom language of the monaco editor, but it was not successful because I needed to customize the sas language rules. I will try using the sas extension of VSCode directly. Thank you again for your help.

**Comment by pipeng119** [comment]
Hey folks, it's already 2025 — has anyone finally implemented SAS syntax highlighting? Just asking if there's a kind soul out there who did it out of love!

**Comment by XiaoPengYouzzZ** [comment]
I ultimately used the local service, opened the VSCode of the server, and implemented this feature with the help of an internal SAS extension @pipeng119
